### PR TITLE
Add envelope fix quirk and value clamping functions

### PR DESCRIPTION
**Pull request by Lawstorant**
This updates the universal quirks branch.

I still think we should try to keep 100% compatibility with pidff so it will be easier to upstream it when we fell like we're ready. I reverted some name changes but of course you can update them after merging if you don't feel like you like them.

I just see no reason to potentially duplicate pidff code as we could just improve it for generic pidff devices.

### CODE REVIEW

File: hid-pidff.c [hunk 1]
```diff
@@ -308,18 +342,14 @@ static void pidff_set_effect_report(struct pidff_device *pidff,
...
 	unsigned short direction = effect->direction;
 
-	if ((effect->type == FF_DAMPER ||
-	    effect->type == FF_FRICTION ||
-	    effect->type == FF_SPRING ||
-	    effect->type == FF_INERTIA) &&
-	    pidff->quirks & PIDFF_QUIRK_FIX_WHEEL_DIRECTION)
-		direction = 0x3FFF;
+	if (pidff->quirks & PIDFF_QUIRK_FIX_WHEEL_DIRECTION)
+		direction = 0x4000;
```
Comment (JacKeTUs) [inline hid-pidff.c:346]: Can you please move that magic number to `#define`? And may be explain in comments why it's exactly that number

**Review by JacKeTUs — CHANGES_REQUESTED** [review]
Just hide magic number into preprocessor, everything else LGTM

File: hid-pidff.c (hunk 1 above)
Comment (Lawstorant) [reply hid-pidff.c]: Done

**Review by JacKeTUs — CHANGES_REQUESTED** [review]

File: dkms.conf [hunk 2]
```diff
@@ -1,8 +1,8 @@
-PACKAGE_NAME="new-pidff"
+PACKAGE_NAME="universal-pidff"
 PACKAGE_VERSION="0.0.2"
 MAKE[0]="make KVERSION=$kernelver"
 CLEAN="make clean"
-BUILT_MODULE_NAME[0]="hid-new-pidff"
-DEST_MODULE_NAME[0]="hid-new-pidff"
+BUILT_MODULE_NAME[0]="hid-univesal-pidff"
```
[the remark this answers is not in the document]
Comment (Lawstorant) [reply dkms.conf]: Oh, well. I was just testing this out and fixed it as well.

[the remark this answers is not in the document]
Comment (Lawstorant) [reply dkms.conf]: Now it works. Confirmed by installing the driver from aur instead of just manually insmoding from repository

**Review by JacKeTUs — APPROVED** [review]

**Pull request merged**